Patch-Source: https://gitlab.gnome.org/GNOME/gthumb/-/merge_requests/32
From d13c10b841ce55251d502193bfb680aa0fa949a5 Mon Sep 17 00:00:00 2001
From: Jakub Jirutka <jakub@jirutka.cz>
Date: Sat, 1 Jan 2022 22:07:05 +0100
Subject: [PATCH] Fix incorrect usage of shared_module in extensions

Meson 0.60.2 prints the following warnings:

    DEPRECATION: target exiv2_tools links against shared module edit_metadata, which is incorrect.
                 This will be an error in the future, so please use shared_library() for edit_metadata instead.
                 If shared_module() was used for edit_metadata because it has references to undefined symbols,
                 use shared_libary() with `override_options: ['b_lundef=false']` instead

This somehow causes that created libimporter.so doesn't have SONAME
and extensions linked with libimporter.so has the following entry
in ELF header:

    0x0000000000000001 (NEEDED)             Shared library: [extensions/importer/libimporter.so]

This is wrong, there should be just libimporter.so (as in all other
extension libs), not extensions/importer/libimporter.so.

Fixes #204

diff --git a/extensions/23hq/meson.build b/extensions/23hq/meson.build
index df33ccd8..da918ae3 100644
--- a/extensions/23hq/meson.build
+++ b/extensions/23hq/meson.build
@@ -5,7 +5,7 @@ source_files = files(
   'main.c'
 )
 
-shared_module('23hq',
+shared_library('23hq',
   sources : source_files,
   dependencies : [
     extension_common_deps,
@@ -19,6 +19,7 @@ shared_module('23hq',
     export_tools_mod,
     importer_mod
   ],
+  override_options : [ 'b_lundef=false' ],
   install : true,
   install_dir : extensions_install_dir,
   install_rpath : extensions_install_dir
diff --git a/extensions/bookmarks/meson.build b/extensions/bookmarks/meson.build
index def6601a..38be8e77 100644
--- a/extensions/bookmarks/meson.build
+++ b/extensions/bookmarks/meson.build
@@ -10,11 +10,12 @@ source_files = files(
 
 gresource_files = gnome.compile_resources('bookmarks-resources', 'resources.xml', c_name : 'gth')
 
-shared_module('bookmarks',
+shared_library('bookmarks',
   sources : [ source_files, gresource_files ],
   dependencies : extension_common_deps,
   include_directories : [ config_inc, gthumb_inc ],
   c_args : c_args,
+  override_options : [ 'b_lundef=false' ],
   install : true,
   install_dir : extensions_install_dir,
   install_rpath : extensions_install_dir
diff --git a/extensions/burn_disc/meson.build b/extensions/burn_disc/meson.build
index a3c750c6..b8a44212 100644
--- a/extensions/burn_disc/meson.build
+++ b/extensions/burn_disc/meson.build
@@ -6,12 +6,13 @@ source_files = files(
   'main.c'
 )
 
-shared_module('burn_disc',
+shared_library('burn_disc',
   sources : source_files,
   dependencies : [ extension_common_deps, libbrasero_dep ],
   include_directories : [ config_inc, gthumb_inc ],
   c_args : c_args,
   link_with : export_tools_mod,
+  override_options : [ 'b_lundef=false' ],
   install : true,
   install_dir : extensions_install_dir,
   install_rpath : extensions_install_dir
diff --git a/extensions/cairo_io/meson.build b/extensions/cairo_io/meson.build
index df6dd8f9..42e4bc6e 100644
--- a/extensions/cairo_io/meson.build
+++ b/extensions/cairo_io/meson.build
@@ -32,7 +32,7 @@ endif
 
 enum_files = gnome.mkenums_simple('cairo-io-enum-types', sources: [ 'preferences.h' ])
 
-shared_module('cairo_io',
+shared_library('cairo_io',
   sources : [ source_files, enum_files ],
   dependencies : [
     extension_common_deps,
@@ -46,6 +46,7 @@ shared_module('cairo_io',
   include_directories : [ config_inc, gthumb_inc ],
   c_args : c_args,
   link_with : jpeg_utils_mod,
+  override_options : [ 'b_lundef=false' ],
   install : true,
   install_dir : extensions_install_dir,
   install_rpath : extensions_install_dir
diff --git a/extensions/catalogs/meson.build b/extensions/catalogs/meson.build
index 24895f66..95dc1f1c 100644
--- a/extensions/catalogs/meson.build
+++ b/extensions/catalogs/meson.build
@@ -10,11 +10,12 @@ source_files = files(
   'main.c'
 )
 
-catalogs_mod = shared_module('catalogs',
+catalogs_mod = shared_library('catalogs',
   sources : source_files,
   dependencies : extension_common_deps,
   include_directories : [ config_inc, gthumb_inc ],
   c_args : c_args,
+  override_options : [ 'b_lundef=false' ],
   install : true,
   install_dir : extensions_install_dir,
   install_rpath : extensions_install_dir
diff --git a/extensions/change_date/meson.build b/extensions/change_date/meson.build
index 4bc427bd..5df517bb 100644
--- a/extensions/change_date/meson.build
+++ b/extensions/change_date/meson.build
@@ -7,11 +7,12 @@ source_files = files(
   'main.c'
 )
 
-shared_module('change_date',
+shared_library('change_date',
   sources : source_files,
   dependencies : extension_common_deps,
   include_directories : [ config_inc, gthumb_inc ],
   c_args : c_args,
+  override_options : [ 'b_lundef=false' ],
   install : true,
   install_dir : extensions_install_dir,
   install_rpath : extensions_install_dir
diff --git a/extensions/comments/meson.build b/extensions/comments/meson.build
index 50bb2289..9371f17d 100644
--- a/extensions/comments/meson.build
+++ b/extensions/comments/meson.build
@@ -8,12 +8,13 @@ source_files = files(
   'main.c'
 )
 
-shared_module('comments',
+shared_library('comments',
   sources : source_files,
   dependencies : [ extension_common_deps ],
   include_directories : [ config_inc, gthumb_inc ],
   c_args : c_args,
   link_with : use_exiv2 ? exiv2_tools_mod : [],
+  override_options : [ 'b_lundef=false' ],
   install : true,
   install_dir : extensions_install_dir,
   install_rpath : extensions_install_dir
diff --git a/extensions/contact_sheet/meson.build b/extensions/contact_sheet/meson.build
index b0d559ab..8639f54e 100644
--- a/extensions/contact_sheet/meson.build
+++ b/extensions/contact_sheet/meson.build
@@ -22,7 +22,7 @@ source_files = files(
   'main.c'
 )
 
-shared_module('contact_sheet',
+shared_library('contact_sheet',
   sources : [ source_files, enum_files ],
   dependencies : [
     extension_common_deps,
@@ -31,6 +31,7 @@ shared_module('contact_sheet',
   include_directories : [ config_inc, gthumb_inc ],
   c_args : c_args,
   link_with: [ image_rotation_mod, export_tools_mod ],
+  override_options : [ 'b_lundef=false' ],
   install : true,
   install_dir : extensions_install_dir,
   install_rpath : extensions_install_dir
diff --git a/extensions/convert_format/meson.build b/extensions/convert_format/meson.build
index 2cdff42a..a807c2cd 100644
--- a/extensions/convert_format/meson.build
+++ b/extensions/convert_format/meson.build
@@ -6,11 +6,12 @@ source_files = files(
   'main.c'
 )
 
-shared_module('convert_format',
+shared_library('convert_format',
   sources : source_files,
   dependencies : extension_common_deps,
   include_directories : [ config_inc, gthumb_inc ],
   c_args : c_args,
+  override_options : [ 'b_lundef=false' ],
   install : true,
   install_dir : extensions_install_dir,
   install_rpath : extensions_install_dir
diff --git a/extensions/desktop_background/meson.build b/extensions/desktop_background/meson.build
index 34558873..773c3bde 100644
--- a/extensions/desktop_background/meson.build
+++ b/extensions/desktop_background/meson.build
@@ -4,12 +4,13 @@ source_files = files(
   'main.c'
 )
 
-shared_module('desktop_background',
+shared_library('desktop_background',
   sources : source_files,
   dependencies : extension_common_deps,
   include_directories : [ config_inc, gthumb_inc ],
   c_args : c_args,
   link_with : image_viewer_mod,
+  override_options : [ 'b_lundef=false' ],
   install : true,
   install_dir : extensions_install_dir,
   install_rpath : extensions_install_dir
diff --git a/extensions/edit_metadata/meson.build b/extensions/edit_metadata/meson.build
index 19114a44..0c86116d 100644
--- a/extensions/edit_metadata/meson.build
+++ b/extensions/edit_metadata/meson.build
@@ -12,11 +12,12 @@ source_files = files(
   'utils.c'
 )
 
-edit_metadata_mod = shared_module('edit_metadata',
+edit_metadata_mod = shared_library('edit_metadata',
   sources : source_files,
   dependencies : extension_common_deps,
   include_directories : [ config_inc, gthumb_inc ],
   c_args : c_args,
+  override_options : [ 'b_lundef=false' ],
   install : true,
   install_dir : extensions_install_dir,
   install_rpath : extensions_install_dir
diff --git a/extensions/exiv2_tools/meson.build b/extensions/exiv2_tools/meson.build
index a4a68a22..2d64194d 100644
--- a/extensions/exiv2_tools/meson.build
+++ b/extensions/exiv2_tools/meson.build
@@ -5,13 +5,14 @@ source_files = files(
   'main.c'
 )
 
-exiv2_tools_mod = shared_module('exiv2_tools',
+exiv2_tools_mod = shared_library('exiv2_tools',
   sources : source_files,
   dependencies : [ extension_common_deps, exiv2_dep ],
   include_directories : [ config_inc, gthumb_inc ],
   c_args : c_args,
   cpp_args : cpp_args,
   link_with : [ edit_metadata_mod ],
+  override_options : [ 'b_lundef=false' ],
   install : true,
   install_dir : extensions_install_dir,
   install_rpath : extensions_install_dir
diff --git a/extensions/export_tools/meson.build b/extensions/export_tools/meson.build
index baac456b..f50ab07a 100644
--- a/extensions/export_tools/meson.build
+++ b/extensions/export_tools/meson.build
@@ -5,11 +5,12 @@ source_files = files(
 
 gresource_files = gnome.compile_resources('export-tools-resources', 'resources.xml', c_name : 'gth')
 
-export_tools_mod = shared_module('export_tools',
+export_tools_mod = shared_library('export_tools',
   sources : [ source_files, gresource_files ],
   dependencies : extension_common_deps,
   include_directories : [ config_inc, gthumb_inc ],
   c_args : c_args,
+  override_options : [ 'b_lundef=false' ],
   install : true,
   install_dir : extensions_install_dir,
   install_rpath : extensions_install_dir
diff --git a/extensions/file_manager/meson.build b/extensions/file_manager/meson.build
index d672162c..b2e90da9 100644
--- a/extensions/file_manager/meson.build
+++ b/extensions/file_manager/meson.build
@@ -7,11 +7,12 @@ source_files = files(
   'main.c'
 )
 
-shared_module('file_manager',
+shared_library('file_manager',
   sources : source_files,
   dependencies : extension_common_deps,
   include_directories : [ config_inc, gthumb_inc ],
   c_args : c_args,
+  override_options : [ 'b_lundef=false' ],
   install : true,
   install_dir : extensions_install_dir,
   install_rpath : extensions_install_dir
diff --git a/extensions/file_tools/meson.build b/extensions/file_tools/meson.build
index 04355025..b292093c 100644
--- a/extensions/file_tools/meson.build
+++ b/extensions/file_tools/meson.build
@@ -75,7 +75,7 @@ source_files = files(
   'main.c'
 )
 
-file_tools_mod = shared_module('file_tools',
+file_tools_mod = shared_library('file_tools',
   sources : [ source_files, enum_files ],
   dependencies : extension_common_deps,
   include_directories : [
@@ -84,6 +84,7 @@ file_tools_mod = shared_module('file_tools',
   ],
   c_args : c_args,
   link_with : image_viewer_mod,
+  override_options : [ 'b_lundef=false' ],
   install : true,
   install_dir : extensions_install_dir,
   install_rpath : extensions_install_dir
diff --git a/extensions/file_viewer/meson.build b/extensions/file_viewer/meson.build
index 50f7ba77..9746271c 100644
--- a/extensions/file_viewer/meson.build
+++ b/extensions/file_viewer/meson.build
@@ -3,11 +3,12 @@ source_files = files(
   'main.c'
 )
 
-shared_module('file_viewer',
+shared_library('file_viewer',
   sources : [ source_files ],
   dependencies : extension_common_deps,
   include_directories : [ config_inc, gthumb_inc ],
   c_args : c_args,
+  override_options : [ 'b_lundef=false' ],
   install : true,
   install_dir : extensions_install_dir,
   install_rpath : extensions_install_dir
diff --git a/extensions/find_duplicates/meson.build b/extensions/find_duplicates/meson.build
index 01e9fa57..7db50827 100644
--- a/extensions/find_duplicates/meson.build
+++ b/extensions/find_duplicates/meson.build
@@ -7,12 +7,13 @@ source_files = files(
   'main.c'
 )
 
-shared_module('find_duplicates',
+shared_library('find_duplicates',
   sources : source_files,
   dependencies : extension_common_deps,
   include_directories : [ config_inc, gthumb_inc ],
   c_args : c_args,
   link_with : catalogs_mod,
+  override_options : [ 'b_lundef=false' ],
   install : true,
   install_dir : extensions_install_dir,
   install_rpath : extensions_install_dir
diff --git a/extensions/flicker/meson.build b/extensions/flicker/meson.build
index 106033c5..06452bff 100644
--- a/extensions/flicker/meson.build
+++ b/extensions/flicker/meson.build
@@ -5,7 +5,7 @@ source_files = files(
   'main.c'
 )
 
-shared_module('flicker',
+shared_library('flicker',
   sources : source_files,
   dependencies : [
     extension_common_deps,
@@ -25,6 +25,7 @@ shared_module('flicker',
     export_tools_mod,
     importer_mod
   ],
+  override_options : [ 'b_lundef=false' ],
   install : true,
   install_dir : extensions_install_dir,
   install_rpath : extensions_install_dir
diff --git a/extensions/flicker_utils/meson.build b/extensions/flicker_utils/meson.build
index 56c6a916..82f6c00c 100644
--- a/extensions/flicker_utils/meson.build
+++ b/extensions/flicker_utils/meson.build
@@ -10,7 +10,7 @@ source_files = files(
   'main.c'
 )
 
-flicker_utils_mod = shared_module('flicker_utils',
+flicker_utils_mod = shared_library('flicker_utils',
   sources : source_files,
   dependencies : [
     extension_common_deps,
@@ -29,6 +29,7 @@ flicker_utils_mod = shared_module('flicker_utils',
     export_tools_mod,
     importer_mod
   ],
+  override_options : [ 'b_lundef=false' ],
   install : true,
   install_dir : extensions_install_dir,
   install_rpath : extensions_install_dir
diff --git a/extensions/gstreamer_tools/meson.build b/extensions/gstreamer_tools/meson.build
index fb60864b..3dfec320 100644
--- a/extensions/gstreamer_tools/meson.build
+++ b/extensions/gstreamer_tools/meson.build
@@ -7,12 +7,13 @@ source_files = files(
   'main.c'
 )
 
-shared_module('gstreamer_tools',
+shared_library('gstreamer_tools',
   sources : source_files,
   dependencies : [ extension_common_deps, gstreamer_deps ],
   include_directories : [ config_inc, gthumb_inc ],
   c_args : c_args,
   link_with: gstreamer_utils_mod,
+  override_options : [ 'b_lundef=false' ],
   install : true,
   install_dir : extensions_install_dir,
   install_rpath : extensions_install_dir
diff --git a/extensions/gstreamer_utils/meson.build b/extensions/gstreamer_utils/meson.build
index a1753738..ca2acea5 100644
--- a/extensions/gstreamer_utils/meson.build
+++ b/extensions/gstreamer_utils/meson.build
@@ -3,11 +3,12 @@ source_files = files(
   'main.c'
 )
 
-gstreamer_utils_mod = shared_module('gstreamer_utils',
+gstreamer_utils_mod = shared_library('gstreamer_utils',
   sources : source_files,
   dependencies : [ extension_common_deps, gstreamer_deps ],
   include_directories : [ config_inc, gthumb_inc ],
   c_args : c_args,
+  override_options : [ 'b_lundef=false' ],
   install : true,
   install_dir : extensions_install_dir,
   install_rpath : extensions_install_dir
diff --git a/extensions/image_print/meson.build b/extensions/image_print/meson.build
index 320b1a8b..8c544734 100644
--- a/extensions/image_print/meson.build
+++ b/extensions/image_print/meson.build
@@ -8,11 +8,12 @@ source_files = files(
   'main.c'
 )
 
-shared_module('image_print',
+shared_library('image_print',
   sources : source_files,
   dependencies : extension_common_deps,
   include_directories : [ config_inc, gthumb_inc ],
   c_args : c_args,
+  override_options : [ 'b_lundef=false' ],
   install : true,
   install_dir : extensions_install_dir,
   install_rpath : extensions_install_dir
diff --git a/extensions/image_rotation/meson.build b/extensions/image_rotation/meson.build
index 9541fc30..0f82d376 100644
--- a/extensions/image_rotation/meson.build
+++ b/extensions/image_rotation/meson.build
@@ -7,12 +7,13 @@ source_files = files(
   'rotation-utils.c'
 )
 
-image_rotation_mod = shared_module('image_rotation',
+image_rotation_mod = shared_library('image_rotation',
   sources : source_files,
   dependencies : [ extension_common_deps, jpeg_deps ],
   include_directories : [ config_inc, gthumb_inc ],
   c_args : c_args,
   link_with : jpeg_utils_mod,
+  override_options : [ 'b_lundef=false' ],
   install : true,
   install_dir : extensions_install_dir,
   install_rpath : extensions_install_dir
diff --git a/extensions/image_viewer/meson.build b/extensions/image_viewer/meson.build
index f8a81615..086cffdb 100644
--- a/extensions/image_viewer/meson.build
+++ b/extensions/image_viewer/meson.build
@@ -12,12 +12,13 @@ source_files = files(
 
 gresource_files = gnome.compile_resources('image-viewer-resources', 'resources.xml', c_name : 'gth')
 
-image_viewer_mod = shared_module('image_viewer',
+image_viewer_mod = shared_library('image_viewer',
   sources : [ source_files, gresource_files ],
   dependencies : extension_common_deps,
   include_directories : [ config_inc, gthumb_inc ],
   c_args : c_args,
   link_with : jpeg_utils_mod,
+  override_options : [ 'b_lundef=false' ],
   install : true,
   install_dir : extensions_install_dir,
   install_rpath : extensions_install_dir
diff --git a/extensions/importer/meson.build b/extensions/importer/meson.build
index 0f779430..5b0c14da 100644
--- a/extensions/importer/meson.build
+++ b/extensions/importer/meson.build
@@ -16,7 +16,7 @@ header_files = [
 
 enum_files = gnome.mkenums_simple('gth-import-enum-types', sources: header_files)
 
-importer_mod = shared_module('importer',
+importer_mod = shared_library('importer',
   sources : [ source_files, enum_files ],
   dependencies : extension_common_deps,
   include_directories : [ config_inc, gthumb_inc ],
@@ -26,6 +26,7 @@ importer_mod = shared_module('importer',
     image_rotation_mod,
     use_exiv2 ? exiv2_tools_mod : []
   ],
+  override_options : [ 'b_lundef=false' ],
   install : true,
   install_dir : extensions_install_dir,
   install_rpath : extensions_install_dir
diff --git a/extensions/jpeg_utils/meson.build b/extensions/jpeg_utils/meson.build
index 77cda188..53c167fc 100644
--- a/extensions/jpeg_utils/meson.build
+++ b/extensions/jpeg_utils/meson.build
@@ -12,7 +12,7 @@ else
   source_files += files('transupp-62.c')
 endif
 
-jpeg_utils_mod = shared_module('jpeg_utils',
+jpeg_utils_mod = shared_library('jpeg_utils',
   sources : source_files,
   dependencies : [
     extension_common_deps,
@@ -20,6 +20,7 @@ jpeg_utils_mod = shared_module('jpeg_utils',
   ],
   include_directories : [ config_inc, gthumb_inc ],
   c_args : c_args,
+  override_options : [ 'b_lundef=false' ],
   install : true,
   install_dir : extensions_install_dir,
   install_rpath : extensions_install_dir
diff --git a/extensions/list_tools/meson.build b/extensions/list_tools/meson.build
index 47becbdb..6414fe7e 100644
--- a/extensions/list_tools/meson.build
+++ b/extensions/list_tools/meson.build
@@ -11,11 +11,12 @@ source_files = files(
 
 gresource_files = gnome.compile_resources('list-tools-resources', 'resources.xml', c_name : 'gth')
 
-list_tools_mod = shared_module('list_tools',
+list_tools_mod = shared_library('list_tools',
   sources : [ source_files, gresource_files ],
   dependencies : extension_common_deps,
   include_directories : [ config_inc, gthumb_inc ],
   c_args : c_args,
+  override_options : [ 'b_lundef=false' ],
   install : true,
   install_dir : extensions_install_dir,
   install_rpath : extensions_install_dir
diff --git a/extensions/map_view/meson.build b/extensions/map_view/meson.build
index 6dacc9a7..45707dfe 100644
--- a/extensions/map_view/meson.build
+++ b/extensions/map_view/meson.build
@@ -3,11 +3,12 @@ source_files = files(
   'main.c'
 )
 
-shared_module('map_view',
+shared_library('map_view',
   sources : source_files,
   dependencies : [ extension_common_deps, libchamplain_deps ],
   include_directories : [ config_inc, gthumb_inc ],
   c_args : c_args,
+  override_options : [ 'b_lundef=false' ],
   install : true,
   install_dir : extensions_install_dir,
   install_rpath : extensions_install_dir
diff --git a/extensions/oauth/meson.build b/extensions/oauth/meson.build
index f6ffc4fb..9d785ac6 100644
--- a/extensions/oauth/meson.build
+++ b/extensions/oauth/meson.build
@@ -10,7 +10,7 @@ source_files = files(
   'web-service.c'
 )
 
-oauth_mod = shared_module('oauth',
+oauth_mod = shared_library('oauth',
   sources : source_files,
   dependencies : [
     extension_common_deps,
@@ -21,6 +21,7 @@ oauth_mod = shared_module('oauth',
   ],
   include_directories : [ config_inc, gthumb_inc ],
   c_args : c_args,
+  override_options : [ 'b_lundef=false' ],
   install : true,
   install_dir : extensions_install_dir,
   install_rpath : extensions_install_dir
diff --git a/extensions/photo_importer/meson.build b/extensions/photo_importer/meson.build
index 4da87100..ca0910f4 100644
--- a/extensions/photo_importer/meson.build
+++ b/extensions/photo_importer/meson.build
@@ -8,7 +8,7 @@ source_files = files(
 
 gresource_files = gnome.compile_resources('photo-importer-resources', 'resources.xml', c_name : 'gth')
 
-photo_importer_mod = shared_module('photo_importer',
+photo_importer_mod = shared_library('photo_importer',
   sources : [ source_files, gresource_files ],
   dependencies : [
     extension_common_deps,
@@ -25,6 +25,7 @@ photo_importer_mod = shared_module('photo_importer',
     importer_mod,
     image_rotation_mod,
   ],
+  override_options : [ 'b_lundef=false' ],
   install : true,
   install_dir : extensions_install_dir,
   install_rpath : extensions_install_dir
diff --git a/extensions/raw_files/meson.build b/extensions/raw_files/meson.build
index 5f3c7d51..6d3e8cfd 100644
--- a/extensions/raw_files/meson.build
+++ b/extensions/raw_files/meson.build
@@ -3,11 +3,12 @@ source_files = files(
   'gth-metadata-provider-raw.c'
 )
 
-shared_module('raw_files',
+shared_library('raw_files',
   sources : source_files,
   dependencies : [ extension_common_deps, libraw_dep ],
   include_directories : [ config_inc, gthumb_inc ],
   c_args : c_args,
+  override_options : [ 'b_lundef=false' ],
   install : true,
   install_dir : extensions_install_dir,
   install_rpath : extensions_install_dir
diff --git a/extensions/red_eye_removal/meson.build b/extensions/red_eye_removal/meson.build
index 52b6f7f5..922cfbab 100644
--- a/extensions/red_eye_removal/meson.build
+++ b/extensions/red_eye_removal/meson.build
@@ -3,7 +3,7 @@ source_files = files(
   'main.c'
 )
 
-shared_module('red_eye_removal',
+shared_library('red_eye_removal',
   sources : source_files,
   dependencies : [
     extension_common_deps,
@@ -19,6 +19,7 @@ shared_module('red_eye_removal',
     file_tools_mod,
     image_viewer_mod
   ],
+  override_options : [ 'b_lundef=false' ],
   install : true,
   install_dir : extensions_install_dir,
   install_rpath : extensions_install_dir
diff --git a/extensions/rename_series/meson.build b/extensions/rename_series/meson.build
index 5c06dc8f..4e1d718a 100644
--- a/extensions/rename_series/meson.build
+++ b/extensions/rename_series/meson.build
@@ -6,11 +6,12 @@ source_files = files(
   'main.c'
 )
 
-shared_module('rename_series',
+shared_library('rename_series',
   sources : source_files,
   dependencies : extension_common_deps,
   include_directories : [ config_inc, gthumb_inc ],
   c_args : c_args,
+  override_options : [ 'b_lundef=false' ],
   install : true,
   install_dir : extensions_install_dir,
   install_rpath : extensions_install_dir
diff --git a/extensions/resize_images/meson.build b/extensions/resize_images/meson.build
index a75eff04..a031b545 100644
--- a/extensions/resize_images/meson.build
+++ b/extensions/resize_images/meson.build
@@ -6,12 +6,13 @@ source_files = files(
   'main.c'
 )
 
-shared_module('resize_images',
+shared_library('resize_images',
   sources : source_files,
   dependencies : extension_common_deps,
   include_directories : [ config_inc, gthumb_inc ],
   c_args : c_args,
   link_with : image_viewer_mod,
+  override_options : [ 'b_lundef=false' ],
   install : true,
   install_dir : extensions_install_dir,
   install_rpath : extensions_install_dir
diff --git a/extensions/search/meson.build b/extensions/search/meson.build
index 7d1879b1..774bb13f 100644
--- a/extensions/search/meson.build
+++ b/extensions/search/meson.build
@@ -10,12 +10,13 @@ source_files = files(
   'main.c'
 )
 
-shared_module('search',
+shared_library('search',
   sources : [ source_files, gresource_files ],
   dependencies : extension_common_deps,
   include_directories : [ config_inc, gthumb_inc ],
   c_args : c_args,
   link_with : catalogs_mod,
+  override_options : [ 'b_lundef=false' ],
   install : true,
   install_dir : extensions_install_dir,
   install_rpath : extensions_install_dir
diff --git a/extensions/selections/meson.build b/extensions/selections/meson.build
index 338c87dc..d4b1cb97 100644
--- a/extensions/selections/meson.build
+++ b/extensions/selections/meson.build
@@ -7,11 +7,12 @@ source_files = files(
   'main.c'
 )
 
-shared_module('selections',
+shared_library('selections',
   sources : source_files,
   dependencies : extension_common_deps,
   include_directories : [ config_inc, gthumb_inc ],
   c_args : c_args,
+  override_options : [ 'b_lundef=false' ],
   install : true,
   install_dir : extensions_install_dir,
   install_rpath : extensions_install_dir
diff --git a/extensions/slideshow/meson.build b/extensions/slideshow/meson.build
index 6f749197..f9819bcd 100644
--- a/extensions/slideshow/meson.build
+++ b/extensions/slideshow/meson.build
@@ -8,7 +8,7 @@ source_files = files(
   'preferences.c'
 )
 
-shared_module('slideshow',
+shared_library('slideshow',
   sources : source_files,
   dependencies : [
     extension_common_deps,
@@ -21,6 +21,7 @@ shared_module('slideshow',
     catalogs_mod,
     use_gstreamer ? gstreamer_utils_mod : []
   ],
+  override_options : [ 'b_lundef=false' ],
   install : true,
   install_dir : extensions_install_dir,
   install_rpath : extensions_install_dir
diff --git a/extensions/terminal/meson.build b/extensions/terminal/meson.build
index df2254c3..d71091ea 100644
--- a/extensions/terminal/meson.build
+++ b/extensions/terminal/meson.build
@@ -5,11 +5,12 @@ source_files = files(
   'main.c'
 )
 
-shared_module('terminal',
+shared_library('terminal',
   sources : source_files,
   dependencies : extension_common_deps,
   include_directories : [ config_inc, gthumb_inc ],
   c_args : c_args,
+  override_options : [ 'b_lundef=false' ],
   install : true,
   install_dir : extensions_install_dir,
   install_rpath : extensions_install_dir
diff --git a/extensions/webalbums/meson.build b/extensions/webalbums/meson.build
index 2519806d..129c379d 100644
--- a/extensions/webalbums/meson.build
+++ b/extensions/webalbums/meson.build
@@ -21,12 +21,13 @@ albumtheme_parser = custom_target('albumtheme_parser',
    depends : albumtheme_lexer
 )
 
-shared_module('webalbums',
+shared_library('webalbums',
   sources : [ source_files, albumtheme_parser ],
   dependencies : extension_common_deps,
   include_directories : [ config_inc, gthumb_inc ],
   c_args : c_args,
   link_with : [ export_tools_mod, image_rotation_mod ],
+  override_options : [ 'b_lundef=false' ],
   install : true,
   install_dir : extensions_install_dir,
   install_rpath : extensions_install_dir
-- 
GitLab

